src/main/java/frc/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,222 @@
+        new TrapezoidProfile.Constraints(
+            ArmConstants.k_maxVelocity, ArmConstants.k_maxAcceleration));
+    setpoint = ArmConstants.k_climbUpSetpoint;
+    // armController.setGoal(ArmConstants.k_climbUpSetpoint);
+    System.out.println("Going to climb up pos");
+  }
+
+  // Moves arm up to shoot note
+  public void goToShootPos() {
+    armController.setConstraints(
+        new TrapezoidProfile.Constraints(
+            ArmConstants.k_maxVelocity, ArmConstants.k_maxAcceleration));
+    setpoint = ArmConstants.k_shootSetpoint;
+    // armController.setGoal(ArmConstants.k_shootSetpoint);
+    System.out.println("Going to shoot pos");
+  }
+
+  public void goToClimbDownPos() {
+    armController.setConstraints(new TrapezoidProfile.Constraints(0.05, 0.05));
+    setpoint = ArmConstants.k_shootSetpoint;
+    // armController.setGoal(ArmConstants.k_shootSetpoint);
+    System.out.println("Going to climb down pos");
+  }
+  // Moves arm to amp position to shoot note
+  double pidVal;
+
+  public void goToAmpPos() {
+    armController.setConstraints(
+        new TrapezoidProfile.Constraints(
+            ArmConstants.k_maxVelocity, ArmConstants.k_maxAcceleration));
+    setpoint = ArmConstants.k_ampSetpoint;
+    System.out.println("Going to amp pos");
+  }
+
+  // Depending on the value of the joysticks, move the arm up or down (manual)
+  public void runArm(double voltage) {
+    if (voltage == 0) {
+      setSpeed(ArmConstants.k_speedZero);
+    } else {
+      if (isClimb) {
+        setSpeed(Math.copySign(ArmConstants.k_climbSpeed, voltage));
+      } else {
+        setSpeed(Math.copySign(ArmConstants.k_armSpeed, voltage));
+      }
+    }
+  }
+
+  // Toggling between climb and regular arm mode because the arm speeds are different for these two
+  public void climbOn() {
+    isClimb = true;
+    // leds.setLEDsRed();
+  }
+
+  public void climbOff() {
+    isClimb = false;
+
+    // leds.setLEDsPurple();
+  }
+
+  private double isAmpUpperBound() {
+    if (isClimb) {
+      return ArmConstants.k_upperBoundNormal;
+    }
+
+    return ArmConstants.k_upperBound;
+  }
+
+  /*public double toggleAmp() {
+  if (isAmp) {
+    return ArmConstants.k_upperBoundAmp;
+  }
+  else {
+    return ArmConstants.k_upperBoundNormal;
+  }*/
+
+  /*public void runClimb() {
+    // setSpeed(Math.copySign(ArmConstants.k_climbSpeed, -1));
+    setSpeed(-0.5);
+  }*/
+
+  // Sets the speed of the motors if it is within the bounds of the robot
+  public void setSpeed(double armSpeed) {
+    if (isInBound(getPosition(), armSpeed) && Math.abs(armSpeed) > 0)
+      armMotor.set(armSpeed); // armMotor.set(armSpeed + setFeedforward())
+    else armMotor.set(0.0);
+  }
+
+  // Calculate the speed to move the arm up and down at the same speed (by taking into account
+  // gravity)
+  public double setFeedforward() {
+    return feedForward.calculate(
+            getPosition().getRadians() + ArmConstants.k_armEncoderOffset.getRadians(), 0.0)
+        / 12.0;
```

Comment:
magic number

Reply:
I believe this was due to the fact that 12V is the maximum the frc battery can take, as can be seen in the docs:
https://docs.wpilib.org/en/stable/docs/hardware/hardware-basics/robot-battery.html#

Therefore, we were trying to convert voltage into speed (if we wanted half-speed which is 0.5, we would do 6.0 / 12, half the total voltage that the motor can have).

Reply:
I've made this a variable in ArmConstants.java and called it in place of the 12.0 magic number.